File: samples/java_layout/src/main/java/io/customer/android/sample/java_layout/sdk/SampleInboxEventListener.java
package io.customer.android.sample.java_layout.sdk;

import androidx.annotation.NonNull;

import java.util.Locale;

import io.customer.android.sample.java_layout.utils.Logger;
import io.customer.messaginginapp.type.InboxActionMessage;
import io.customer.messaginginapp.type.InboxEventListener;

/**
 * Sample implementation for {@link InboxEventListener}.
 * <p>
 * Observational: it logs each callback and returns {@code false} from
 * {@link #messageActionTaken} so the SDK still applies its default action handling (e.g. opening an
 * http(s) url). Return {@code true} instead to fully handle the action and suppress that default.
 */
public class SampleInboxEventListener implements InboxEventListener {
    private final Logger logger;

    public SampleInboxEventListener(Logger logger) {
        this.logger = logger;
    }

    @Override
    public boolean messageActionTaken(@NonNull InboxActionMessage message, @NonNull String actionName, @NonNull String actionValue) {
        logEvent("messageActionTaken. name: %s, value: %s, message: %s", actionName, actionValue, message.getMessageId());
        return false;
    }

    @Override
    public void messageShown(@NonNull InboxActionMessage message) {
        logEvent("messageShown. message: %s", message.getMessageId());
    }

    @Override
    public void messageOpened(@NonNull InboxActionMessage message) {
        logEvent("messageOpened. message: %s", message.getMessageId());
    }

    @Override
    public void messageDismissed(@NonNull InboxActionMessage message) {
        logEvent("messageDismissed. message: %s", message.getMessageId());
    }

    private void logEvent(@NonNull String format, @NonNull Object... args) {
        logger.v(String.format(Locale.ENGLISH, "[CIO-Inbox] sample listener: " + format, args));
    }
}

File: samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/data/sdk/SampleInboxEventListener.kt
package io.customer.android.sample.kotlin_compose.data.sdk

import io.customer.android.sample.kotlin_compose.util.Logger
import io.customer.messaginginapp.gist.data.model.InboxMessage
import io.customer.messaginginapp.type.InboxEventListener

/**
 * Sample implementation for `InboxEventListener`.
 *
 * Observational: it logs each callback and returns `false` from [messageActionTaken] so the SDK
 * still applies its default action handling (e.g. opening an http(s) url). Return `true` instead to
 * fully handle the action and suppress that default.
 */
class SampleInboxEventListener(private val logger: Logger = Logger()) : InboxEventListener {

    override fun messageActionTaken(
        message: InboxMessage,
        actionName: String,
        actionValue: String
    ): Boolean {
        logEvent("messageActionTaken. name: $actionName, value: $actionValue, message: ${message.queueId}")
        return false
    }

    override fun messageShown(message: InboxMessage) {
        logEvent("messageShown. message: ${message.queueId}")
    }

    override fun messageOpened(message: InboxMessage) {
        logEvent("messageOpened. message: ${message.queueId}")
    }

    override fun messageDismissed(message: InboxMessage) {
        logEvent("messageDismissed. message: ${message.queueId}")
    }

    private fun logEvent(message: String) {
        logger.v("[CIO-Inbox] sample listener: $message")
    }
}
